src/content/docs/en/guides/content-collections.mdx

```diff
@@ -261,26 +331,64 @@ Your blog post content here.
 
 ## Querying Collections
 
-Astro provides two functions to query a collection and return one (or more) content entries: [`getCollection()`](/en/reference/api-reference/#getcollection) and [`getEntryBySlug()`](/en/reference/api-reference/#getentrybyslug).
+Astro provides two functions to query a collection and return one (or more) content entries: [`getCollection()`](/en/reference/api-reference/#getcollection) and [`getEntry()`](/en/reference/api-reference/#getentry).
 
 ```js
-import { getCollection, getEntryBySlug } from 'astro:content';
-// Get all entries from a collection. Requires the name of the collection as an argument.
+import { getCollection, getEntry } from 'astro:content';
+
+// Get all entries from a collection.
+// Requires the name of the collection as an argument.
+// Example: retrieve `src/content/blog/**`
 const allBlogPosts = await getCollection('blog');
-// Get a single entry from a collection. Requires the name of the collection and the entry's slug as arguments.
-const oneBlogPost = await getEntryBySlug('blog', 'enterprise');
+
+// Get a single entry from a collection.
+// Requires the name of the collection and either
+// the entry `slug` (content collections) or `id` (data collections)
+// Example: retrieve `src/content/authors/grace-hopper.json`
+const graceHopperProfile = await getEntry('authors', 'grace-hopper');
 ```
 
 Both functions return content entries as defined by the [`CollectionEntry`](/en/reference/api-reference/#collection-entry-type) type.
 
-#### Filtering collection queries
+### Resolving references
+
+When using [collection references](#defining-collection-references), you can use the `getEntry()` and `getEntries()` functions to retrieve the referenced entry `data`. 
```

Comment:
```suggestion
### Using an entry's referenced data

Any [references defined in your schema](#defining-collection-references) must be queried separately after first querying your collection entry. You can use the `getEntry()` function again, or `getEntries()`, to retrieve the referenced entry from the returned `data` object. 
```
Ok, what do you think about something more along the lines of this? I feel like being more explicit about a "2-step query approach" might be helpful? Happy for you to adjust as necessary!

Possible headings? (Note, not just for SEO, but for someone scanning the page TOC and thinking of words that resonate: "Yes! That's what I'm trying to do!")
> Using an entry's referenced data
> Querying an entry's references
> Accessing referenced data

Reply:
I like "Accessing referenced data" the most 🏆